src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -355,6 +355,10 @@ export class CellModel extends Disposable implements ICellModel {
 		if (this.cellType !== CellTypes.Code) {
 			return;
 		}
+
+		// There can only be one tagged parameters cell in the Notebook
+		value = this.notebookModel?.cells?.find(cell => cell.isParameter) ? false : value;
```

Comment:
Note here - in general this kind of design is best avoided. If I call a setter and give it a value I would expect it to always be set. Silently "ignoring" the value passed in is unintuitive for the caller and can lead to unexpected behavior.

Two better options I see : 

1. Move this to a function and make it clear that it isn't always going to actually update the value (maybe return a bool indicating success, such as with the tryX methods you see a lot in C#)
1. Throw an error if an existing cell is a parameter

Throwing an error seems like a good choice - since it generally seems like it'd be an error if we had multiple cells with the parameter tag (given the implementation and usage of the tag).

At the very least make sure this behavior is documented in the doc comments to hopefully avoid callers not handling this correctly.

Reply:
Will look into throwing an error when there is already a cell tagged as parameters.